majit/majit-ir/src/intbound.rs

```
@@ -0,0 +1,2611 @@
+        IntBound::bounded(lower, upper)
+    }
+
+    /// Bound after left shift.
+    pub fn lshift_bound(&self, other: &IntBound) -> IntBound {
+        let (mut tvalue, mut tmask) = (0u64, u64::MAX); // TNUM_UNKNOWN
+        if other.is_constant() {
+            let c_other = other.get_constant_int();
+            if c_other >= 64 {
+                tvalue = 0;
+                tmask = 0; // TNUM_KNOWN_ZERO
+            } else if c_other >= 0 && c_other < 64 {
```

Clippy (1.95.0)

[warning] manual `Range::contains` implementation

(warning)